## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#42bc9c90-f629-44af-ae5c-ed53cd894605) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#31c48f41-22c2-4ffa-ae6b-60e234749c09) |
| mint-pools.sh | :x: [more...](#942be28f-bd5d-4572-a602-5d3fa09c7211) |
<a name="942be28f-bd5d-4572-a602-5d3fa09c7211"></a>
### 15887-96cf279/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.5:31644
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp eb7692f2c2ca:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 minutes and 32 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-v95pbjkpy10zp1",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 211738,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.5:31644/aws-sdk-go-test-v95pbjkpy10zp1\": dial tcp 15.15.15.5:31644: i/o timeout",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-v95pbjkpy10zp1",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 211738,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.5:31644/aws-sdk-go-test-v95pbjkpy10zp1\": dial tcp 15.15.15.5:31644: i/o timeout",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 2 minutes and 11 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-v95pbjkpy10zp1",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 211738,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.5:31644/aws-sdk-go-test-v95pbjkpy10zp1\": dial tcp 15.15.15.5:31644: i/o timeout",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 36 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 32147.13,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 11 seconds
{
  "name": "awscli",
  "duration": 9252,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... FAILED in 30 seconds
{
  "alert": "",
  "args": null,
  "duration": 30000,
  "error": "Get \"http://15.15.15.5:31644/minio/prometheus/metrics\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testPrometheusEndpoint",
  "message": "GET request to Prometheus endpoint failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 10 seconds
{
  "alert": "",
  "args": null,
  "duration": 30000,
  "error": "Get \"http://15.15.15.5:31644/minio/prometheus/metrics\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testPrometheusEndpoint",
  "message": "GET request to Prometheus endpoint failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-go tests ... FAILED in 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-eyijv9ymxp4jc9ja",
    "region": "eu-west-1"
  },
  "duration": 3829,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(1/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 229,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-3uqufr1, objectName = null, resource = /minio-java-test-3uqufr1, requestId = 171F2761C435F63C, hostId = ca6902b6-9140-42fc-9d75-208d14405c65)\nrequest={method=PUT, url=http://15.15.15.5:31644/minio-java-test-3uqufr1, headers=Host: 15.15.15.5:31644\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221018T114807Z\nAuthorization: ██\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 338\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171F2761C435F63C\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 18 Oct 2022 11:48:08 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(1/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(1/14) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-a042f3ac-6c6d-4159-b17e-0d2465f9a76c",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.5', port=31644): Max retries exceeded with url: /minio-py-test-a042f3ac-6c6d-4159-b17e-0d2465f9a76c (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.5', port=31644): Max retries exceeded with url: /minio-py-test-a042f3ac-6c6d-4159-b17e-0d2465f9a76c (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600493
}
(1/14) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45367",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(1/14) Running s3select tests ... FAILED in 12 minutes and 12 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-36132e4b-35ee-4831-aa19-759ee426b05e"
  },
  "duration": 730946,
  "message": "HTTPConnectionPool(host='15.15.15.5', port=31644): Max retries exceeded with url: /s3select-test-36132e4b-35ee-4831-aa19-759ee426b05e (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34a11974f0>: Failed to establish a new connection: [Errno 110] Connection timed out'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nTimeoutError: [Errno 110] Connection timed out\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f34a11974f0>: Failed to establish a new connection: [Errno 110] Connection timed out\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.5', port=31644): Max retries exceeded with url: /s3select-test-36132e4b-35ee-4831-aa19-759ee426b05e (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34a11974f0>: Failed to establish a new connection: [Errno 110] Connection timed out'))\n",
  "status": "FAIL"
}
(1/14) Running versioning tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-t1annejiu9ut3o"
  },
  "duration": 364902,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F28F7D791CB8B, host id: ",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="31c48f41-22c2-4ffa-ae6b-60e234749c09"></a>
### 15887-96cf279/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.19:30590
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 33bf5f5a49c4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 4 minutes and 33 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5ncykbf0reyjix",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 272442,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F273B31B0E09E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 12 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5ncykbf0reyjix",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 272442,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F273B31B0E09E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5ncykbf0reyjix",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 272442,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F273B31B0E09E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 21 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 17132.36,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 1 minutes and 10 seconds
{
  "name": "awscli",
  "duration": 68481,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... FAILED in 30 seconds
{
  "alert": "",
  "args": null,
  "duration": 30029,
  "error": "Get \"http://15.15.15.19:30590/minio/health/live\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 30029,
  "error": "Get \"http://15.15.15.19:30590/minio/health/live\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-go tests ... FAILED in 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-d44zc1cpxcuuhcit",
    "region": "eu-west-1"
  },
  "duration": 2863,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(1/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 227,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-1apif1m, objectName = null, resource = /minio-java-test-1apif1m, requestId = 171F275B64F21444, hostId = null)\nrequest={method=PUT, url=http://15.15.15.19:30590/minio-java-test-1apif1m, headers=Host: 15.15.15.19:30590\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.5\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20221018T114740Z\nAuthorization: ██\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 302\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171F275B64F21444\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 18 Oct 2022 11:47:40 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(1/14) Running minio-js tests ... FAILED in 2 minutes and 11 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 22,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(1/14) Running minio-py tests ... FAILED in 10 minutes and 25 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-5e91b668-3877-47ad-b69b-ee06922012c6",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.19', port=30590): Max retries exceeded with url: /minio-py-test-5e91b668-3877-47ad-b69b-ee06922012c6 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.19', port=30590): Max retries exceeded with url: /minio-py-test-5e91b668-3877-47ad-b69b-ee06922012c6 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 625413
}
(1/14) Running s3cmd tests ... FAILED in 2 minutes and 55 seconds
{
  "name": "s3cmd",
  "duration": "175513",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / ([Errno 110] Connection timed out)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(1/14) Running s3select tests ... FAILED in 10 minutes and 25 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-482efbd8-4a37-4163-a064-60c97d8753c1"
  },
  "duration": 624721,
  "message": "HTTPConnectionPool(host='15.15.15.19', port=30590): Max retries exceeded with url: /s3select-test-482efbd8-4a37-4163-a064-60c97d8753c1 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.19', port=30590): Max retries exceeded with url: /s3select-test-482efbd8-4a37-4163-a064-60c97d8753c1 (Caused by ResponseError('too many 503 error responses'))\n",
  "status": "FAIL"
}
(1/14) Running versioning tests ... FAILED in 4 minutes and 33 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-6uoj9rv465y1lu"
  },
  "duration": 272826,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F29057B870817, host id: ",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="42bc9c90-f629-44af-ae5c-ed53cd894605"></a>
### 15887-96cf279/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.8:31124
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 8b0c19bafca4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 4 minutes and 35 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-x5soztwktehu6a",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 274863,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F273BF1E499C2, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 22 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-x5soztwktehu6a",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 274863,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F273BF1E499C2, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-x5soztwktehu6a",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 274863,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F273BF1E499C2, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2118.02,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 8 seconds
{
  "name": "awscli",
  "duration": 7277,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... FAILED in 30 seconds
{
  "alert": "",
  "args": null,
  "duration": 30001,
  "error": "Get \"http://15.15.15.8:31124/minio/prometheus/metrics\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testPrometheusEndpoint",
  "message": "GET request to Prometheus endpoint failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 30001,
  "error": "Get \"http://15.15.15.8:31124/minio/prometheus/metrics\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
  "function": "testPrometheusEndpoint",
  "message": "GET request to Prometheus endpoint failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-go tests ... FAILED in 1 minutes and 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-whxiais0pg1ka54o",
    "region": "eu-west-1"
  },
  "duration": 63804,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(1/14) Running minio-java tests ... FAILED in 2 minutes and 12 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 130566,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to /15.15.15.8:31124 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(1/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(1/14) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-52b64212-c3df-4397-ac2e-b885786ca0b8",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.8', port=31124): Max retries exceeded with url: /minio-py-test-52b64212-c3df-4397-ac2e-b885786ca0b8 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.8', port=31124): Max retries exceeded with url: /minio-py-test-52b64212-c3df-4397-ac2e-b885786ca0b8 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600546
}
(1/14) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45362",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(1/14) Running s3select tests ... FAILED in 10 minutes and 10 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-b2f4d0e9-2f4a-43cd-8bb3-84837d900969"
  },
  "duration": 609997,
  "message": "HTTPConnectionPool(host='15.15.15.8', port=31124): Max retries exceeded with url: /s3select-test-b2f4d0e9-2f4a-43cd-8bb3-84837d900969 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.8', port=31124): Max retries exceeded with url: /s3select-test-b2f4d0e9-2f4a-43cd-8bb3-84837d900969 (Caused by ResponseError('too many 503 error responses'))\n",
  "status": "FAIL"
}
(1/14) Running versioning tests ... FAILED in 6 minutes and 7 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-cvb21yorhe56hj"
  },
  "duration": 366123,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 171F28F36B7A48BE, host id: ",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/common-main.go
+++ b/cmd/common-main.go
@@ -175,7 +175,9 @@ func minioConfigToConsoleFeatures() {
 	if globalMinioEndpoint != "" {
 		os.Setenv("CONSOLE_MINIO_SERVER", globalMinioEndpoint)
 	} else {
-		os.Setenv("CONSOLE_MINIO_SERVER", getAPIEndpoints()[0])
+		// Explicitly set 127.0.0.1 so Console will automatically bypass TLS verification to the local S3 API.
+		// This will save users from providing a certificate with IP or FQDN SAN that points to the local host.
+		os.Setenv("CONSOLE_MINIO_SERVER", fmt.Sprintf("%s://127.0.0.1:%s", getURLScheme(globalIsTLS), globalMinioPort))
 	}
 	if value := env.Get("MINIO_LOG_QUERY_URL", ""); value != "" {
 		os.Setenv("CONSOLE_LOG_QUERY_URL", value)

--- a/cmd/gateway-main.go
+++ b/cmd/gateway-main.go
@@ -249,7 +249,7 @@ func StartGateway(ctx *cli.Context, gw Gateway) {
 		getCert = globalTLSCerts.GetCertificate
 	}
 
-	httpServer := xhttp.NewServer([]string{globalMinioAddr}).
+	httpServer := xhttp.NewServer(getServerListenAddrs()).
 		UseHandler(setCriticalErrorHandler(corsHandler(router))).
 		UseTLSConfig(newTLSConfig(getCert)).
 		UseShutdownTimeout(ctx.Duration("shutdown-timeout")).

--- a/cmd/net.go
+++ b/cmd/net.go
@@ -44,9 +44,8 @@ func mustSplitHostPort(hostPort string) (host, port string) {
 	return xh.Name, xh.Port.String()
 }
 
-// mustGetLocalIP4 returns IPv4 addresses of localhost.  It panics on error.
-func mustGetLocalIP4() (ipList set.StringSet) {
-	ipList = set.NewStringSet()
+// mustGetLocalIPs returns IPs of local interface
+func mustGetLocalIPs() (ipList []net.IP) {
 	ifs, err := net.Interfaces()
 	logger.FatalIf(err, "Unable to get IP addresses of this host")
 
@@ -68,36 +67,33 @@ func mustGetLocalIP4() (ipList set.StringSet) {
 				ip = v.IP
 			}
 
-			if ip.To4() != nil {
-				ipList.Add(ip.String())
-			}
+			ipList = append(ipList, ip)
 		}
 	}
 
 	return ipList
 }
 
+// mustGetLocalIP4 returns IPv4 addresses of localhost.  It panics on error.
+func mustGetLocalIP4() (ipList set.StringSet) {
+	ipList = set.NewStringSet()
+	for _, ip := range mustGetLocalIPs() {
+		if ip.To4() != nil {
+			ipList.Add(ip.String())
+		}
+	}
+	return
+}
+
 // mustGetLocalIP6 returns IPv6 addresses of localhost.  It panics on error.
 func mustGetLocalIP6() (ipList set.StringSet) {
 	ipList = set.NewStringSet()
-	addrs, err := net.InterfaceAddrs()
-	logger.FatalIf(err, "Unable to get IP addresses of this host")
-
-	for _, addr := range addrs {
-		var ip net.IP
-		switch v := addr.(type) {
-		case *net.IPNet:
-			ip = v.IP
-		case *net.IPAddr:
-			ip = v.IP
-		}
-
+	for _, ip := range mustGetLocalIPs() {
 		if ip.To4() == nil {
 			ipList.Add(ip.String())
 		}
 	}
-
-	return ipList
+	return
 }
 
 // getHostIP returns IP address of given host.

--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -25,6 +25,7 @@ import (
 	"io"
 	"log"
 	"math/rand"
+	"net"
 	"os"
 	"os/signal"
 	"runtime"
@@ -36,6 +37,7 @@ import (
 	"github.com/minio/cli"
 	"github.com/minio/minio-go/v7"
 	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/minio/minio-go/v7/pkg/set"
 	"github.com/minio/minio/internal/auth"
 	"github.com/minio/minio/internal/bucket/bandwidth"
 	"github.com/minio/minio/internal/color"
@@ -418,6 +420,24 @@ func initConfigSubsystem(ctx context.Context, newObject ObjectLayer) error {
 	return nil
 }
 
+// Return the list of address that MinIO server needs to listen on:
+//   - Returning 127.0.0.1 is necessary so Console will be able to send
+//     requests to the local S3 API.
+//   - The returned List needs to be deduplicated as well.
+func getServerListenAddrs() []string {
+	// Use a string set to avoid duplication
+	addrs := set.NewStringSet()
+	// Listen on local interface to receive requests from Console
+	for _, ip := range mustGetLocalIPs() {
+		if ip != nil && ip.IsLoopback() {
+			addrs.Add(net.JoinHostPort(ip.String(), globalMinioPort))
+		}
+	}
+	// Add the interface specified by the user
+	addrs.Add(globalMinioAddr)
+	return addrs.ToSlice()
+}
+
 // serverMain handler called for 'minio server' command.
 func serverMain(ctx *cli.Context) {
 	signal.Notify(globalOSSignalCh, os.Interrupt, syscall.SIGTERM, syscall.SIGQUIT)
@@ -500,7 +520,7 @@ func serverMain(ctx *cli.Context) {
 		getCert = globalTLSCerts.GetCertificate
 	}
 
-	httpServer := xhttp.NewServer([]string{globalMinioAddr}).
+	httpServer := xhttp.NewServer(getServerListenAddrs()).
 		UseHandler(setCriticalErrorHandler(corsHandler(handler))).
 		UseTLSConfig(newTLSConfig(getCert)).
 		UseShutdownTimeout(ctx.Duration("shutdown-timeout")).

--- a/cmd/signals.go
+++ b/cmd/signals.go
@@ -75,7 +75,8 @@ func handleSignals() {
 
 	for {
 		select {
-		case <-globalHTTPServerErrorCh:
+		case err := <-globalHTTPServerErrorCh:
+			logger.LogIf(context.Background(), err)
 			exit(stopProcess())
 		case osSignal := <-globalOSSignalCh:
 			if !globalIsGateway {

